RelationalRecord/src/Relations/User.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, FlexibleInstances #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module  Relations.User where
+
+import  Types.User as User
+import  Types.DB
+import  Relations.DB
+
+import  Database.Relational.Query
+import  Database.HDBC.Query.TH      (makeRecordPersistableDefault)
+
+import  Data.Int                    (Int32)
+
+
+
+-- SELECTS
+
+allUsers :: Relation () Users
+allUsers = User.users
+
+getUser :: Relation PKey Users
+getUser = relation' . placeholder $ \usrId -> do
+    a       <- query users
+    wheres  $ a ! User.id' .=. usrId
+    return  a
+
+
+-- INSERTS
+
+data UserInsert = UserInsert
+    { iTenantId     :: PKey
+    , iUsername     :: String
+    , iPassword     :: String
+    , iFirstName    :: Maybe String
+    , iLastName     :: Maybe String
+    }
+$(makeRecordPersistableDefault ''UserInsert)
+
+piUser :: Pi Users UserInsert
+piUser = UserInsert
+    |$| User.tenantId'
+    |*| User.username'
+    |*| User.password'
+    |*| User.firstName'
+    |*| User.lastName'
+
+insertUser :: Insert UserInsert
+insertUser = derivedInsert piUser
+
+
+-- UPDATES
+
+data UserUpdate = UserUpdate
```

Comment:
how do we handle the case where certain columns need to be protected from "blind" updates? eg. password column? While it can be inserted initially, it should not be updatable later via a variadic update? One should be forced to call a special function for changing the password, which would also trigger a side-effect of notifying the user by email?